tests/dataset_builders/pie/test_lisa.py

```diff
@@ -0,0 +1,368 @@
+import dataclasses
+from collections import defaultdict
+from typing import List, Optional, Union
+
+import pytest
+from datasets import disable_caching
+from pie_datasets import DatasetDict
+from pie_datasets.builders.brat import BratDocument, BratDocumentWithMergedSpans
+from pie_modules.document.processing import tokenize_document
+from pytorch_ie.annotations import BinaryRelation, LabeledMultiSpan, LabeledSpan
+from pytorch_ie.core import AnnotationList, annotation_field
+from pytorch_ie.documents import TextDocumentWithLabeledSpansAndBinaryRelations, TokenBasedDocument
+from transformers import AutoTokenizer, PreTrainedTokenizer
+
+from dataset_builders.pie.lisa.lisa import Lisa
+from tests.dataset_builders import PIE_BASE_PATH
+
+
+@dataclasses.dataclass
+class TestTokenDocumentWithLabeledSpans(TokenBasedDocument):
+    labeled_spans: AnnotationList[LabeledSpan] = annotation_field(target="tokens")
+
+
+@dataclasses.dataclass
+class TestTokenDocumentWithLabeledSpansAndBinaryRelations(TestTokenDocumentWithLabeledSpans):
+    binary_relations: AnnotationList[BinaryRelation] = annotation_field(target="labeled_spans")
+
+
+disable_caching()
+
+DATASET_NAME = "lisa"
+PIE_DATASET_PATH = PIE_BASE_PATH / DATASET_NAME
+SPLIT_SIZES = {"train": 118}
+
+
+@pytest.fixture(scope="module", params=["default", "merge_fragmented_spans"])
+def dataset_variant(request) -> str:
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def dataset(dataset_variant) -> DatasetDict:
+    return DatasetDict.load_dataset(str(PIE_DATASET_PATH), name=dataset_variant)
+
+
+def test_dataset(dataset):
+    assert dataset is not None
+    assert {name: len(ds) for name, ds in dataset.items()} == SPLIT_SIZES
+
+
+@pytest.fixture(scope="module")
+def document(dataset) -> Union[BratDocument, BratDocumentWithMergedSpans]:
+    return dataset["train"][0]
+
+
+def test_document(document, dataset_variant):
+    assert document is not None
+    assert document.text.startswith(
+        "Hallo liebe <user>, danke für deine lieben Worte, hier im Forum sind alle so nett und hilfsbereit."
+    )
+    if dataset_variant == "default":
+        assert isinstance(document, BratDocument)
+        span = document.spans[0]
+        assert isinstance(span, LabeledMultiSpan)
+        # TODO
+    elif dataset_variant == "merge_fragmented_spans":
+        assert isinstance(document, BratDocumentWithMergedSpans)
+        span = document.spans[0]
+        assert isinstance(span, LabeledSpan)
+        # TODO
+    else:
+        raise ValueError(f"Unknown dataset variant: {dataset_variant}")
+
+
+def my_converter(document: BratDocument) -> TextDocumentWithLabeledSpansAndBinaryRelations:
+    spans = document.spans
+    new_doc = TextDocumentWithLabeledSpansAndBinaryRelations(text=document.text)
```

Comment:
I recommend to also set `id=document.id` and `metadata=copy.deepcopy(document.metadata)`.

Reply:
So in metadata, we have information regarding the span slices, do we need to split them for instance, [[[34, 40], [50, 58]], [[41, 49]], [[76, 80]],...] to [[[34, 40]], [[50, 58]], [[41, 49]], [[76, 80]],...] and corresponding text ?

Reply:
no, I would not put any effort into that. In general, I just pass all existing metadata to all newly created documents in the case that may be needed later on.